planner/core/logical_plan_builder.go

```diff
@@ -2290,6 +2292,15 @@ func (b *PlanBuilder) popTableHints() {
 	b.tableHintInfo = b.tableHintInfo[:len(b.tableHintInfo)-1]
 }
 
+func (b *PlanBuilder) appendUnmatchedIndexHintWarning(indexHints []indexHintInfo) {
+	for _, hint := range indexHints {
+		if !hint.matched {
+			errMsg := fmt.Sprintf("IndexHint for table %s is not applicable.", hint.tblName)
```

Comment:
Any suggestion for the warning message? cc @eurekaka @zz-jason

Reply:
how about:
```sql
fmt.Sprintf(
	"%s(%d.%s, %s) is inapplicable, check whether the table(%s.%s) or index(%s) exists",
	hint.indexHint.HintType,
	hint.dbName,
	hint.tblName,
	...
)
```

Reply:
Why not directly call the `ast.HintTable.restore`?

Reply:
@lzmhhh123 `indexHint` is `ast.IndexHint`, it doesn't contain `DB` or `Table` when use `restore`.